src/main/java/frc/robot/subsystems/Vision.java

```diff
@@ -123,36 +126,37 @@ public Optional<EstimatedRobotPose> getEstimatedGlobalPose(Pose2d prevEstimatedR
         return m_photonPoseEstimator.update();
     }
 
-	/*public Pose2d getTagPose() {
-		var targetResult = m_aprilTagCamera.getLatestResult();
-		
-		int targetID = -1;
-		if (targetResult.hasTargets()) {
-			// Get the current best target.
-			PhotonTrackedTarget target = targetResult.getBestTarget();
-			targetID = target.getFiducialId();
-		}
-		return m_aprilTagFieldLayout.getTagPose(targetID).get().toPose2d();
-
-	}*/
-    //returns ID of best target
-    public int getBestTagID(){
+    public Pose2d getCentralTagPose() { // gets target closets to center of camera
         var targetResult = m_aprilTagCamera.getLatestResult();
-		
-		int targetID = -1;
-		if (targetResult.hasTargets()) {
-			// Get the current best target.
-			PhotonTrackedTarget target = targetResult.getBestTarget();
-			targetID = target.getFiducialId();
-		}
-        return targetID; 
+        int targetID = -1;
+        if (targetResult.hasTargets()) {
+            // Get lists of targets
+            List<PhotonTrackedTarget> targets = targetResult.getTargets();
+            // make a temp holder var for least Y translation
+            Translation2d translationHolder = new Translation2d(0, 0);
+            for (PhotonTrackedTarget tag : targets) { // for every target in camera
+                // get transformation to target
+                Transform3d tagTransform = tag.getBestCameraToTarget();
+                // get translation to target from transformation
+                Translation2d tagTranslation = tagTransform.getTranslation().toTranslation2d();
+
+                // looking for smallest absolute relative to camera Y
+                // if abs Y translation of new tag is less then holder tag, it becomes holder tag
+                if (Math.abs(tagTranslation.getY()) < Math.abs(translationHolder.getY())) {
+                    translationHolder = tagTranslation;
+                    targetID = tag.getFiducialId(); // set targetID
+                }
+            }
+        }
+        return m_aprilTagFieldLayout.getTagPose(targetID).get().toPose2d();
```

Comment:
This can cause NullPointerException if the there is no target in the camera and targetID remains -1, which is not a valid tag ID in the layout. It will throw NullPointerException when you try to make it call .toPose2d()

You should handle this case somehow. It seems like this pose will be passed to getTagRobotPose() and then used in trajectoryToPose(). I guess one way to fix this is use the Optional<Pose2d> that the library uses a lot, and check if there is a value present in it or not when constructing the trajectoryToPose(). Or I guess add an if statements to check if the variable is null in each of these three places